yateto/input.py

```diff
@@ -104,9 +105,16 @@ def parseJSONMatrixFile(jsonFile, clones=dict(), transpose=lambda name: False, a
     content = json.load(j)
     for m in content:
       entries = m['entries']
-      if len(next(iter(entries))) == 2:
-        entries = [(entry[0], entry[1], True) for entry in entries]
-      matrices.update( __processMatrix(m['name'], m['rows'], m['columns'], entries, clones, transpose, alignStride, namespace) )
+      if 'rows' in m:
+        shape = [m['rows']]
+        if 'columns' in m:
+          shape += [m['columns']]
+      else:
+        shape = m['shape']
+      dim = len(shape)
+      if len(next(iter(entries))) == dim:
```

Comment:
next(iter(entries))) will be always the first item of the iteration. Is that what you want?

Reply:
As far as I'm aware (it's from the previous code), it's supposed to do exactly that:

* an entry of length `dim` will only encode a sparsity pattern
* an entry of length `(dim+1)` will encode a value as well

(it's not a very ideal solution, arguably; at least if we consider the special case of null matrices)